src/main/groovy/net/neoforged/gradleutils/spotless/SpotlessPlugin.groovy

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.gradleutils.spotless
+
+import groovy.transform.CompileStatic
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+
+@CompileStatic
+class SpotlessPlugin implements Plugin<Project> {
+    @Override
+    void apply(Project project) {
+        project.extensions.create('spotlessUtils', SpotlessUtilsExtension, project)
```

Comment:
We should either apply the Spotless plugin ourselves, or adjust this so the extension is only created if/when the Spotless plugin is applied.

Reply:
We cannot apply the spotless plugin because we'd need to depend on it.

Reply:
Then, we should be able to use [`PluginManager#withPlugin`](https://docs.gradle.org/8.4/dsl/org.gradle.api.plugins.PluginManager.html#org.gradle.api.plugins.PluginManager:withPlugin(java.lang.String,%20org.gradle.api.Action)) to register the extension when Spotless is applied (and no sooner), at least.